src/blockchain/log-processors/order-created.ts

```diff
@@ -56,8 +56,20 @@ export function processOrderCreatedLog(db: Knex, augur: Augur, trx: Knex.Transac
           worseOrderID: log.worseOrderID,
           tradeGroupID: log.tradeGroupID,
         };
-        db.transacting(trx).insert(dataToInsert).into("orders").asCallback(callback);
+        const orderID = { orderID: log.orderId };
+        trx.select(["marketID"]).from("orders").where(orderID).asCallback((err: Error|null, ordersRows?: any): void => {
+          if (err) return callback(err);
+          if (!ordersRows || !ordersRows.length) {
+            db.transacting(trx).insert(Object.assign(orderData, orderID)).into("orders").asCallback(callback);
+          } else {
+            db.transacting(trx).from("orders").where(orderID).update(orderData).asCallback(callback);
+          }
+        });
       });
     });
   });
 }
+
+export function processOrderCreatedLogRemoval(db: Knex, augur: Augur, trx: Knex.Transaction, log: FormattedLog, callback: ErrorCallback): void {
+  processOrderCanceledLog(db, augur, trx, log, callback);
```

Comment:
Interesting, so we're not actually removing a created order during a reorg, we're just marking it as cancelled, but when it comes along to be recreated later, the insert-or-update logic above will do the right thing? The artifact here is that an order will exist with `isRemoved: 1`, but that seems harmless

Reply:
Yep, that seemed to be the easiest way to deal with the whole "but if we cancel it and the cancel gets removed, we no longer have the order info so what do we do :(" problem!